components/Buttons/src/MDCButton.m

```diff
@@ -819,7 +819,11 @@ - (void)updateTitleFont {
 
   if (_mdc_adjustsFontForContentSizeCategory) {
     // Dynamic type is enabled so apply scaling
-    font = [font mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleButton scaledForDynamicType:YES];
+    if (font.mdc_scalingCurve && !_mdc_legacyFontScaling) {
+      font = [font mdc_scaledFontForCurrentSizeCategory];
```

Comment:
is there no need to provide what style to give the font?

Reply:
We do a little earlier in the Sample.
This font is set by the themer, which takes it from containerScheme.typographyScheme.button.

In the case of the new scheme, that font will have an mdc_scalingCurve attached.

Reply:
ah awesome